### PR TITLE
Updated IdServiceImpl

### PR DESCRIPTION
**Pull request by vipuldhurve02**

### CODE REVIEW

File: authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/impl/IdServiceImpl.java [hunk 1]
```diff
@@ -327,9 +327,15 @@ private void updateVIDstatus(String vid) throws IdAuthenticationBusinessExceptio
 			vid = securityManager.hash(vid);
 			// Assumption : If transactionLimit is null, id is considered as Perpetual VID
 			// If transactionLimit is nonNull, id is considered as Temporary VID
+
 			if (identityRepo.existsById(vid)
 					&& Objects.nonNull(identityRepo.getOne(vid).getTransactionLimit())) {
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/impl/IdServiceImpl.java]: Please replace code duplication for `identityRepo.getOne(vid).getTransactionLimit()` by assigning it into one variable.

File: authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/impl/IdServiceImpl.java [hunk 2]
```diff
@@ -327,9 +327,15 @@ private void updateVIDstatus(String vid) throws IdAuthenticationBusinessExceptio
...
 			// Assumption : If transactionLimit is null, id is considered as Perpetual VID
 			// If transactionLimit is nonNull, id is considered as Temporary VID
+
 			if (identityRepo.existsById(vid)
 					&& Objects.nonNull(identityRepo.getOne(vid).getTransactionLimit())) {
-				identityRepo.deleteById(vid);
+				int transactionLimit = identityRepo.getOne(vid).getTransactionLimit();
+
+				if(transactionLimit>0){
+					identityRepo.getOne(vid).setTransactionLimit(transactionLimit-1);
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/impl/IdServiceImpl.java]: Please remove code duplication for `identityRepo.getOne(vid)`

File: authentication/authentication-common/src/test/java/io/mosip/authentication/common/service/impl/IdServiceImplTest.java [hunk 3]
```diff
@@ -298,6 +297,10 @@ public void updateVidStatusTest() throws IdAuthenticationBusinessException {
 		Mockito.when(identityRepo.existsById(vid)).thenReturn(true);
 		Mockito.when(identityRepo.getOne(vid)).thenReturn(entity);
 		ReflectionTestUtils.invokeMethod(idServiceImpl, "updateVIDstatus", vid);
+
+		entity.setTransactionLimit(0);
+		Mockito.when(identityRepo.getOne(vid)).thenReturn(entity);
+		ReflectionTestUtils.invokeMethod(idServiceImpl, "updateVIDstatus", vid);
 	}
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/test/java/io/mosip/authentication/common/service/impl/IdServiceImplTest.java:377]: Please add test cases for below:
* transacstion limit is 1
* transaction limit is 2
* transaction limit is -1
* transaction limit is null

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/impl/IdServiceImpl.java [hunk 4]
```diff
@@ -327,9 +327,15 @@ private void updateVIDstatus(String vid) throws IdAuthenticationBusinessExceptio
...
+				}
+				else identityRepo.deleteById(vid);
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/main/java/io/mosip/authentication/common/service/impl/IdServiceImpl.java]: Make the else statement as block using {}

File: authentication/authentication-common/src/test/java/io/mosip/authentication/common/service/impl/IdServiceImplTest.java (hunk 3 above)
Comment (vipuldhurve02) [reply authentication/authentication-common/src/test/java/io/mosip/authentication/common/service/impl/IdServiceImplTest.java:377]: I've added all the above cases.

File: authentication/authentication-common/src/test/java/io/mosip/authentication/common/service/impl/IdServiceImplTest.java [hunk 5]
```diff
@@ -290,27 +285,27 @@ public void getTokenTest() {
 	@Test
 	public void updateVidStatusTest() throws IdAuthenticationBusinessException {
 		String vid = "213131";
-		IdentityEntity entity = getEntity();
-		entity.setTransactionLimit(2);
+		Optional<IdentityEntity> entityOpt = Optional.of(getEntity());
```
Comment (loganathan-sekaran) [inline authentication/authentication-common/src/test/java/io/mosip/authentication/common/service/impl/IdServiceImplTest.java:295]: Split each use case as separate test method, giving proper name to the test method as per use case. Add assertion about the transaction limit and delete call..

**Review by loganathan-sekaran — CHANGES_REQUESTED** [review]

File: authentication/authentication-common/src/test/java/io/mosip/authentication/common/service/impl/IdServiceImplTest.java (hunk 5 above)
Comment (vipuldhurve02) [reply authentication/authentication-common/src/test/java/io/mosip/authentication/common/service/impl/IdServiceImplTest.java:295]: Added assertion as well as created separate test methods

**Review by loganathan-sekaran — APPROVED** [review]
Approved.

**Pull request merged**